charts/argo-cd/values.yaml

```diff
@@ -65,6 +65,12 @@ global:
     # -- Set the global logging level. One of: `debug`, `info`, `warn` or `error`
     level: info
 
+  # -- Annotations for the all deployed Statefulsets
+  statefulsetAnnotations: {}
```

Comment:
Is the type important or simple `annotations` would be enough? Technically all things in this chart are Deployments and only controller component is StatefulSet.

Reply:
While type isn't important, elsewhere in this chart is the value concept of podAnnotations.  As such, since it was accepted, I presumed this was a desired standard of distinguish that which you were annotating.  Therefore, I carried forward that convention.  Reworking to dissolve that convention is something theoretically doable... but it feels like scope creep against what this PR is trying to solve... no?

If you disagree, perhaps a more general convention could be adopted throughout.  Maybe a generalized "annotations" key with subkeys "pod", "deployment", and "statefulset"?  Even if only one of the statefulsets are present in the entire helm chart, it is, and it's not a choice I made... and therefore one for which I have a use case that must account that I suspect others may have as well.

Reply:
1) I'm asking because If you take a look on every single resource in this chart (services, accounts, secrets, configmaps, etc.) you will see `annotations` and `labels` as a standard for metadata section. Yet the PR is using nonstandard `controller.deploymentAnnotations` for metadata section. This is would also imply that controller with `kind: StatefulSet` is actually a deployment which is not true.

2) For global section I'm wondering if globally defined common annotations should be split based on component kind and what is the actual real use case for that.

Reply:
For point the first, I do not believe you are correct.  From current README, here's all the Helm chart values I can find that allow custom annotations:

`
crds.annotations | object | {} | Annotations to be added to all CRDs
global.podAnnotations | object | {} | Annotations for the all deployed pods
configs.cm.annotations | object | {} | Annotations to be added to argocd-cm configmap
-- | -- | -- | --
`

Reply:
It is inaccurate to say "every single resource in this chart":

| global.podAnnotations | object | `{}` | Annotations for the all deployed pods |
| configs.credentialTemplatesAnnotations | object | `{}` | Annotations to be added to `configs.credentialTemplates` Secret |
| configs.repositoriesAnnotations | object | `{}` | Annotations to be added to `configs.repositories` Secret |
| controller.podAnnotations | object | `{}` | Annotations to be added to application controller pods |
| repoServer.podAnnotations | object | `{}` | Annotations to be added to repo server pods |
| server.podAnnotations | object | `{}` | Annotations to be added to server pods |
| dex.podAnnotations | object | `{}` | Annotations to be added to the Dex server pods |
| redis.podAnnotations | object | `{}` | Annotations to be added to the Redis server pods |
| applicationSet.podAnnotations | object | `{}` | Annotations for the controller pods |
| notifications.podAnnotations | object | `{}` | Annotations to be applied to the controller Pods |

The "actual real use case for that", in this case, is third party software suites that operate based on said annotations that I cannot presently use with your helm chart.  For example, Stakater Reloader https://github.com/stakater/Reloader.  There are secrets in the helm chart that the software does not reload the pods for when required to do so.  As such, our organizational strategy is to leverage this whenever applications have edge cases where Secrets data they use do not reload upon that data's change when required to.  Any extra Secret I add to related Deployments or Statefulsets will operate this way.  The only one I know of that you auto reload is SSL cert for argocd-server, and we've even had that fail on us once (cert-manager updated the Secret for the TLS cert, argo-cd did not reload it until pod termination and restart... that will be a separate issue with our findings there).

As for global, I'll rephrase my first response on the matter.  I wouldn't be opposed to, say:

global:
  annotations:
    pod:
    statefulset:
    deployment:

If we did this, though, global.podAnnotations would then become redundant, and, naturally, for consistency, you'd want to rip that out.  Will you require this of me to approve this PR?

Reply:
Thanks for explaining this and you are right that there are some places where this convention isn't followed. I was trying to figure why are new specific definitions for annotations required so we can avoid unnecessary tech debt.